datasource/src/main/scala/quasar/physical/mongo/MongoConfig.scala

```diff
@@ -20,14 +20,58 @@ import slamdata.Predef._
 
 import argonaut._, Argonaut._
 
+import cats.kernel.Order
+
 import com.mongodb.ConnectionString
 
 import quasar.physical.mongo.MongoResource.{Database, Collection}
 
+trait PushdownLevel extends Product with Serializable
+final case object Disabled extends PushdownLevel
```

Comment:
We've been defining case constructors in the parent type's companion object by convention.